### PR TITLE
GFORMS-2746 - Add ability to test for duplicates in Add To List component

### PR DESCRIPTION
**Pull request by cmanson**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/formtemplate/BooleanExprValidator.scala [hunk 1]
```diff
@@ -80,6 +81,25 @@ class BooleanExprValidator(
...
+            Valid
+          }
+          .getOrElse(
+            Invalid(
+              s"id '${f.formComponentId}' named in $wrapperType expression does not exist in an Add To List component"
+            )
+          )
+      )
+      .toList ++ fcIds.map(validateIdIdx)
+  }
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/formtemplate/BooleanExprValidator.scala:101]: Can we move this logic out of `BooleanExprValidator`? Despite its name this class should be used only for checking forward references.

Checking of invalid references is done in `validateInvalidReferences` of `FormTemplateValidator`

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/formtemplate/BooleanExprValidator.scala (hunk 1 above)
Comment (cmanson) [reply app/uk/gov/hmrc/gform/formtemplate/BooleanExprValidator.scala]: Hi @josef-vlach, can you please review the updates I've made based on your review?

File: app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala [hunk 2]
```diff
@@ -169,6 +169,43 @@ object FormTemplateValidator {
...
+
+        val verifiedIncludeIf: List[ValidationResult] =
+          page.includeIf
+            .map(_.booleanExpr)
+            .map(verifyIncludeIf.apply)
+            .getOrElse(List(Valid))
+
+        verifiedValidIfs ++ verifiedComponentIncludeIfs ++ verifiedRemoveItemIf ++ verifiedIncludeIf
+      }
+
```
[the remark this answers is not in the document]
Comment (cmanson) [reply app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala]: Hi Josef, can you review this again now please?

**Review by josef-vlach — APPROVED** [review]
